### PR TITLE
Add admin course server

### PR DESCRIPTION
**Pull request by ebauman**
**What this PR does / why we need it**:
- Adds admin course server
- Resolves a few code bugaboos in the non-admin course server

**Which issue(s) this PR fixes**: n/a

### CODE REVIEW

**Comment by ebauman** [comment]
Related: https://github.com/hobbyfarm/admin-ui/pull/49

File: pkg/admin/courseserver/courseserver.go [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+	id := vars["id"]
+
+	if id == "" {
+		util.ReturnHTTPMessage(w, r, 400, "badrequest", "no id passed in")
+		return
+	}
+
+	obj, err := a.hfClientSet.HobbyfarmV1().Courses().Get(id, metav1.GetOptions{})
+	if err != nil {
+		util.ReturnHTTPMessage(w, r, 400,"badrequest", "invalid id")
```
Comment (Oats87) [inline pkg/admin/courseserver/courseserver.go:63]: nit: missing space between `400,` and `"badrequest"`

File: pkg/admin/courseserver/courseserver.go [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+	encodedCourse, err := json.Marshal(preparedCourse)
+	if err != nil {
+		glog.Errorf("error encoding course: %v", err)
+		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error retrieving course")
+		return
+	}
+
+	util.ReturnHTTPContent(w, r, 200, "success", encodedCourse)
+
+	glog.V(2).Infof("retrieve course %s", obj.Name)
```
Comment (Oats87) [inline pkg/admin/courseserver/courseserver.go:78]: is this too low of a log level to be logging this type of info? I'd say we can make this `4`

File: pkg/admin/courseserver/courseserver.go [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+	encodedCourses, err := json.Marshal(courses)
+	if err != nil {
+		glog.Errorf("error marshalling prepared courses: %v", err)
+		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error listing courses")
+		return
+	}
+
+	util.ReturnHTTPContent(w, r, 200, "success", encodedCourses)
+
+	glog.V(2).Infof("listed courses")
```
Comment (Oats87) [inline pkg/admin/courseserver/courseserver.go:109]: is this too low of a log level to be logging this type of info? I'd say we can make this `4`

File: pkg/admin/courseserver/courseserver.go [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+		return
+	}
+
+	description := r.PostFormValue("description")
+	if description == "" {
+		util.ReturnHTTPMessage(w, r, 400, "badrequest", "no description passed in")
+		return
+	}
+
+	keepaliveDuration := r.PostFormValue("keepalive_duration")
```
Comment (Oats87) [inline pkg/admin/courseserver/courseserver.go:132]: just as a side thought, I wonder if we should validate these values against malicious users.

File: pkg/admin/courseserver/courseserver.go [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+	}
+
+	course, err = a.hfClientSet.HobbyfarmV1().Courses().Create(course)
+	if err != nil {
+		glog.Errorf("error creating course %v", err)
+		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error creating course")
+		return
+	}
+
+	util.ReturnHTTPMessage(w, r, 201, "created", course.Name)
```
Comment (Oats87) [inline pkg/admin/courseserver/courseserver.go:192]: can we add a log statement ( >= `4`) that we created a Course?

File: pkg/admin/courseserver/courseserver.go [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+		_, updateErr := a.hfClientSet.HobbyfarmV1().Courses().Update(course)
+		return updateErr
+	})
+
+	if retryErr != nil {
+		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error attempting to update")
+		return
+	}
+
+	util.ReturnHTTPMessage(w, r, 200, "updated", "")
```
Comment (Oats87) [inline pkg/admin/courseserver/courseserver.go:288]: can we add a log statement ( >= 4) that we updated a Course?

File: pkg/admin/courseserver/courseserver.go [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+
+	err = a.hfClientSet.HobbyfarmV1().Courses().Delete(id, &metav1.DeleteOptions{})
+	if err != nil {
+		glog.Errorf("error deleting course: %v", err)
+		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error while deleting course")
+		return
+	}
+
+	util.ReturnHTTPMessage(w, r, 204, "deleted", "deleted successfully")
+	glog.V(2).Infof("deleted course: %v", id)
```
Comment (Oats87) [inline pkg/admin/courseserver/courseserver.go:362]: I think this is too low of a log level.

File: pkg/admin/courseserver/courseserver.go (hunk 4 above)
Comment (ebauman) [reply pkg/admin/courseserver/courseserver.go:132]: i.e. regex them?

Comment (Oats87) [reply pkg/admin/courseserver/courseserver.go:132]: regex would be one way to do it, and probably safe enough™

Comment (ebauman) [reply pkg/admin/courseserver/courseserver.go:132]: do you want that part of this commit, or handled as part of a project-wide strategy to regex

Comment (ebauman) [reply pkg/admin/courseserver/courseserver.go:132]: e.g. new PR specifically for that

Comment (Oats87) [reply pkg/admin/courseserver/courseserver.go:132]: @ebauman that's up to you. probably a separate commit makes sense, I was just making a comment

**Pull request merged**